### PR TITLE
[stable/ambassador] var for scoping to namespace

### LINKED ISSUE #12669: [stable/ambassador]better control for namespace scope
Currently you can set `env.AMBASSADOR_SINGLE_NAMESPACE` to tell ambassador to only watch resources in the current namespace. But if you have a chain of dependent charts then once one turns this on then you can't turn it off again. This is because ambassador [expects the var to either be set or not set and unsetting a value in a map is problematic](https://github.com/SeldonIO/seldon-core/pull/445#issuecomment-477953912).

It would be smoother to make this its own variable wrapped in an if statement, as it[ used to be at some point](https://github.com/datawire/ambassador/pull/657). 

For other env vars it is fine to use the env map but this one is better handled by wrapping in an if statement as [ambassador works by looking at whether it is set or not set](https://github.com/datawire/ambassador/issues/588#issuecomment-407671801), not by looking at the value of the var

### PR DESCRIPTION
**Pull request by ryandawsonuk**
fixes https://github.com/helm/charts/issues/12669 by effectively re-applying https://github.com/datawire/ambassador/pull/657

Using the term 'scope' as I think that makes it clearer that it sets what ambassador will watch. More like https://github.com/helm/charts/blob/d2a6482b3e374a52d8dbb904a6004dfe99231b86/incubator/aws-alb-ingress-controller/values.yaml#L30 and https://github.com/helm/charts/blob/cbd5e811a44c7bac6226b019f1d1810ef5ee45fa/incubator/istio/templates/custom-resources/mixer.yaml#L16

### CODE REVIEW

**Comment by kflynn** [comment]
@Flydiverny thoughts? I'm generally OK with making the Ambassador chart more Helm-like, but we've shifted some other stuff deliberately to the environment recently, and this is shifting single-namespace away from the environment...

File: stable/ambassador/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 0.52.0
 description: A Helm chart for Datawire Ambassador
 name: ambassador
-version: 2.0.1
+version: 2.0.2
```
Comment (Flydiverny) [inline stable/ambassador/Chart.yaml]: Can we set this as `2.1.0` instead

```suggestion
version: 2.1.0
```

**Review by Flydiverny — CHANGES_REQUESTED** [review]

**Comment by Flydiverny** [comment]
Rebase and bump version as suggested and test should prob be ok :)

**Pull request merged**